autoPyTorch/data/utils.py

```diff
@@ -0,0 +1,309 @@
+# Implementation used from https://github.com/automl/auto-sklearn/blob/development/autosklearn/util/data.py
+import warnings
+from math import floor
+from typing import (
+    Any,
+    Dict,
+    Iterator,
+    List,
+    Mapping,
+    Optional,
+    Sequence,
+    Tuple,
+    Type,
+    Union,
+    cast
+)
+
+import numpy as np
+
+import pandas as pd
+from pandas.api.types import is_float_dtype, is_numeric_dtype
+
+from scipy.sparse import issparse, spmatrix
+
+
+# TODO: TypedDict with python 3.8
+#
+#   When upgrading to python 3.8 as minimum version, this should be a TypedDict
+#   so that mypy can identify the fields types
+DatasetCompressionSpec = Dict[str, Union[int, float, List[str]]]
+DatasetDTypeContainerType = Union[Type, Dict[str, Type]]
+DatasetCompressionInputType = Union[np.ndarray, spmatrix, pd.DataFrame]
```

Comment:
Oh, I did not know that we have something like `spmatrix`...
Probably I should publish a PR to replace typing by this.

Reply:
sure, but its not a priority.